src/api/initialDataTypes.js

```diff
@@ -324,11 +347,18 @@ export type InitialDataRealm = $ReadOnly<{|
   server_generation: number,
   server_inline_image_preview: boolean,
   server_inline_url_embed_preview: boolean,
+
+  // TODO(server-8.0): Added in feat. 212
+  server_jitsi_server_url?: string | null,
+
   server_name_changes_disabled: boolean,
 
   // TODO(server-5.0): Added in feat. 74
   server_needs_upgrade?: boolean,
 
+  // TODO(server-8.0): Added in feat. 221
+  server_supported_permission_settings?: JSONableDict, // unstable
```

Comment:
Reading through the changelog, there's also these at FL 164:

> [POST /register](https://zulip.com/api/register-queue): Added the server_presence_ping_interval_seconds and server_presence_offline_threshold_seconds fields for clients to use when implementing the [presence](https://zulip.com/api/get-presence) system.

Reply:
And similarly, at FL 204:

> [POST /register](https://zulip.com/api/register-queue): Added server_typing_started_wait_period_milliseconds, server_typing_stopped_wait_period_milliseconds, and server_typing_started_expiry_period_milliseconds fields for clients to use when implementing [typing notifications](https://zulip.com/api/set-typing-status) protocol.

Reply:
Hmm. The commit is supposed to be about specifically updating `InitialDataRealm`, and those fields aren't designated

> Present if `realm` is present in `fetch_event_types`.

. Do you think that's an accident, like the four accidents I reported on [CZO](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.60.2Fregister.60.3A.20fields.20missing.20.22Present.20if.20.5B.E2.80.A6.5D.22.3F/near/1707782)?

Also, these are addressed in 47a8b3fd2 and 45951db11, respectively.

Reply:
> those fields aren't designated
> 
> > Present if `realm` is present in `fetch_event_types`.
> 
> . Do you think that's an accident, like the four accidents I reported on [CZO](https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.60.2Fregister.60.3A.20fields.20missing.20.22Present.20if.20.5B.E2.80.A6.5D.22.3F/near/1707782)?

Looks like it must be — in zerver/lib/events.py, if you search for those names you find they get set in the `if want("realm"):` block.

Which has this comment:
```py
    if want("realm"):
        # The realm bundle includes both realm properties and server
        # properties, since it's rare that one would want one and not
        # the other. We expect most clients to want it.
```
(It goes on from there, but that's the part relevant to this question.)


> Also, these are addressed in [47a8b3f](https://github.com/zulip/zulip-mobile/commit/47a8b3fd26b39adc272f059b7fe2ce5cbcff1188)

Hmm, yeah, so I put those in `InitialDataBase`. But given the above, they actually belong in `InitialDataRealm`.


> and [45951db](https://github.com/zulip/zulip-mobile/commit/45951db11e7888cf7d42b1753de95806f669e860), respectively.

I'd describe that more as skipping addressing it :slightly_smiling_face: More precisely, we skipped there actually looking at these fields (and propagating them through the reducer into `RealmState`), so as a result we didn't bother adding them to the API type either.

Here, since we're bumping the "This is current to …" note, it'd be good to add these types along the way in order to keep that note accurate. I think the comments in 45951db11e7888cf7d42b1753de95806f669e860 don't even need any adjustment for that.

Reply:
Cool; this all makes sense. I've noted the fields missing the "if `realm` is present" note in the relevant CZO discussion: https://chat.zulip.org/#narrow/stream/412-api-documentation/topic/.60.2Fregister.60.3A.20fields.20missing.20.22Present.20if.20.5B.E2.80.A6.5D.22.3F/near/1712821